account/rethinkdb/store.go

```diff
@@ -0,0 +1,279 @@
+package rethinkdb
+
+import (
+	"fmt"
+
+	rdb "github.com/dancannon/gorethink"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/go-errors/errors"
+	"github.com/ory-am/common/pkg"
+	"github.com/ory-am/hydra/account"
+	"github.com/ory-am/hydra/hash"
+	"github.com/pborman/uuid"
+)
+
+const accountsTable = "hydra_account"
+
+type Store struct {
+	hasher  hash.Hasher
+	session *rdb.Session
+}
+
+func New(h hash.Hasher, session *rdb.Session) *Store {
+	return &Store{h, session}
+}
+
+func (s *Store) CreateTables() error {
+	exists, err := s.tableExists(accountsTable)
+	if err == nil && !exists {
+		_, err := rdb.TableCreate(accountsTable).RunWrite(s.session)
+		if err != nil {
+			fmt.Println(err)
+		}
+	}
+	return nil
+}
+
+// TableExists : check if table(s) exists in database
+func (s *Store) tableExists(table string) (bool, error) {
+
+	res, err := rdb.TableList().Run(s.session)
+	if err != nil {
+		return false, err
+	}
+	defer res.Close()
+
+	if res.IsNil() {
+		return false, nil
+	}
+
+	var tableDB string
+	for res.Next(&tableDB) {
+		if table == tableDB {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func validate(r interface{}) error {
+	if v, err := govalidator.ValidateStruct(r); !v {
+		return pkg.ErrInvalidPayload
+	} else if err != nil {
+		return pkg.ErrInvalidPayload
+	}
+	return nil
+}
+
+func (s *Store) Create(r account.CreateAccountRequest) (account.Account, error) {
+	var err error
+
+	if r.ID == "" {
```

Comment:
seeing this I'm considering wether this should be handled by each database implementation or the calee instead..what do you think?